velox/exec/fuzzer/JoinFuzzer.cpp

```diff
@@ -1093,6 +1213,32 @@ void JoinFuzzer::verify(core::JoinType joinType) {
     }
   }
 
+  const auto tableScanDir = exec::test::TempDirectoryPath::create();
+
+  // Test cross product without filter with 10% chance. Avoid testing cross
+  // product if input size is too large.
+  if ((core::isInnerJoin(joinType) || core::isLeftJoin(joinType) ||
+       core::isFullJoin(joinType)) &&
+      FLAGS_batch_size * FLAGS_num_batches <= 500) {
+    if (vectorFuzzer_.coinToss(0.1)) {
+      testCrossProduct(
+          tableScanDir->getPath(),
+          joinType,
+          probeKeys,
+          buildKeys,
+          probeInput,
+          buildInput);
+      testCrossProduct(
+          tableScanDir->getPath(),
+          joinType,
+          probeKeys,
+          buildKeys,
+          flatProbeInput,
+          flatBuildInput);
+      return;
```

Comment:
Why we want to return here?

Reply:
testCrossProduct() executes a nested loop join with no join condition. Hash join and merge join cannot execute join with no join condition, so there is no need to continue executing alternative join plans for this iteration.

Reply:
Then shall we remove core::isInnerJoin(joinType) || core::isLeftJoin(joinType) in condition above? Thanks!